test/spectests/spectests.cpp

```diff
@@ -289,6 +296,11 @@ class test_runner
                     pass();
```

Comment:
I think actually we should check the `type` here. It is a pass in `assert_malformed`, but a fail in `assert_invalid`.